xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/js/xwiki/editors/dataeditors.js

```diff
@@ -692,40 +700,40 @@
           });
         });
         let self = this;
-        Sortable.create('xclassContent', {
-          tag: 'div',
-          only: 'xproperty',
-          handle: 'move',
-          starteffect: self.startDrag,
-          endeffect: self.endDrag,
-          onUpdate: self.updateOrder
+        $('#xclassContent').sortable({
+          cursor: 'move',
+          items: '.xproperty',
+          start: self.startDrag,
+          stop: self.endDrag,
+          update: self.updateOrder
         });
       }
-    },
-    updateOrder : function(container) {
-      let i = 0;
-      $(container).children().each(function () {
-        $(this).find(".xproperty-content").data('numberProperty', i++);
-      });
-    },
-    startDrag : function(dragged) {
-      $(dragged).addClass('dragged');
+    }
+
+    updateOrder() {
+      let i = 1;
+      $(this).find(".xproperty-content").data('numberProperty').val(i++);
```

Comment:
In this case it probably sets the same value to all the elements in the jQuery collection, so I think you need to keep passing a function so that the value is reevaluated for each element.